### ISSUE TITLE
work needed to allow users to run analyses separate to pathogen workflows

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/pathogen-repo-guide#74]
We recently had a general discussion ([meeting notes, 2024-11-21/22](https://docs.google.com/document/d/1eENEFFE2oQr4phQwLpToqJ4KLKOn5_3XqtfF_ggplYI/edit?tab=t.0#heading=h.1z06pkr557gt)) about what would be required for workflows to be runnable by external collaborators entirely from custom configs, i.e. no modification of the Snakefile. Here's a summary of what we discussed, specifically as it pertains to this guide. I don't think we're actually very far away from achieving this, and the aim of this summary is to help guide us to this goal and add things that we consider 

- [ ] **Running workflows from a separate analysis directory** There are prototypes of this [here](https://github.com/nextstrain/measles/pull/55) and [here](https://github.com/nextstrain/avian-flu/pull/103), and the corresponding CLI prototype is [here](https://github.com/nextstrain/cli/pull/407). Once these prototypes have solidified I expect the necessary snakemake modifications will be added to this guide. While they still require the workflow repo directory to be managed by the user (git cloned, updated etc), I think that's enough for the purposes of this goal; if we can get the CLI to manage them then even better.

- [ ] **Adding private metadata & sequences to workflows (https://github.com/nextstrain/pathogen-repo-guide/issues/72)**  This is blocked on https://github.com/nextstrain/augur/issues/1579 but there's no reason we can't trial this out for metadata-only additions with `augur merge` right now. If we consider private metadata curation beyond the scope of this (I do) then we already have working script-based approaches (ncov, mpox) to follow.

- [ ] **Allow customisation of DTA columns etc** This guide doesn't specify the config style & snakemake rules for these parts of the workflow so in practice they'll most likely be copied from existing workflows. I think these are config-customisable by setting your own `config["traits"]["columns"]` (or similar). The desired customisations brought up in the recent meeting were all conceptually similar to this, they didn't involve toggling rules on/off or other more complex changes to the workflow. So perhaps we'd consider this task already done?

- [ ] **Workflow versioning**. When running analyses separate to the workflow it's crucial to make it clear when user configs are out-of-date and provide a path to updating them. Conversely it's desired to know what effect a particular config value has, although that seems a harder problem to me and perhaps docs + config validation would achieve this. We can start by using [one-off checks within code](https://github.com/nextstrain/avian-flu/pull/104/commits/d79f0b46172b9196ace6c8b14eab9377bef28a01#diff-47959dfd378b3cd1d39b5515418ee8e4444ab7a6036d5197a5bea82814f928a3R183-R184)  however [linters](https://github.com/nextstrain/pathogen-repo-guide/issues/28) and config schemas would be more powerful¹. We've talked forever about generating [docs from schemas](https://github.com/nextstrain/augur/issues/819) and perhaps that's a direction we could take for pathogen repos from day 1. 

¹ My experience with schemas in augur is that they are good at identifying invalid data but poor at explaining what's wrong and therefore hint at how to fix it. Presumably the schemas here will be simpler so the error messages may be more informative.

**Comment by genehack** [comment]
Bit of a nitpick, but I think an important one in terms of problem scoping and definition. When you say: 

> what would be required for workflows to be runnable by external collaborators entirely from custom configs, i.e. no modification of the Snakefile.

…would it not be more accurate to say, "i.e., no modification of _anything in the pathogen repo._"?

**Comment by huddlej** [comment]
Regarding "Running workflows from a separate analysis directory", [this analysis of seasonal flu data from Loes et al. 2024](https://github.com/blab/loes-2024-neutralization-nextstrain/) is another example of how I've tried to setup an analysis that uses [the seasonal-flu workflow](https://github.com/nextstrain/seasonal-flu) but that lives outside of that repo as another external user might.

**Comment by tsibley** [comment]
@huddlej :+1: That's what the [ncov tutorial](https://docs.nextstrain.org/projects/ncov/en/latest/tutorial/example-data.html#setup) does too.

**Comment by victorlin** [comment]
ncov tutorial is based on similar usage by the old `my_profiles` directory, reflected in [ncov's .gitignore](https://github.com/nextstrain/ncov/blame/52edcc2179bc2cf7b8e8dfc0f4d483f8934f0541/.gitignore#L13-L19)
